geniusyield-market-maker/src/GeniusYield/MarketMaker/Spread.hs

```diff
@@ -0,0 +1,39 @@
+module GeniusYield.MarketMaker.Spread (Spread (..)) where
+
+import           Control.Applicative           ((<|>))
+import           Data.Aeson                    (FromJSON (..))
+import qualified Data.Aeson                    as Aeson
+import           Deriving.Aeson
+import           GeniusYield.MarketMaker.Utils (camelToSnake)
+
+data Spread = Spread
+  { buySideSpread  :: !Rational,
+    sellSideSpread :: !Rational
+  }
+  deriving stock (Show, Generic)
+  deriving (ToJSON) via CustomJSON '[FieldLabelModifier '[CamelToSnake]] Spread
+
+-- |
+--
+-- >>> Aeson.decode @Spread "{\"numerator\":41,\"denominator\":10000}"
+-- Just (Spread {buySideSpread = 41 % 10000, sellSideSpread = 41 % 10000})
+--
+-- >>> Aeson.encode $ Spread 0.01 0.02
+-- "{\"buy_side_spread\":{\"numerator\":1,\"denominator\":100},\"sell_side_spread\":{\"numerator\":1,\"denominator\":50}}"
+--
+instance FromJSON Spread where
+  parseJSON v =
+    Aeson.genericParseJSON (Aeson.defaultOptions {Aeson.fieldLabelModifier = camelToSnake}) v
+      <|> ( do
+              n ← parseJSON v
+              pure $ Spread n n
+          )
+
+instance Num Spread where
```

Comment:
`Num` instance should be correct as I see other folks also define similarly: https://github.com/conal/NumInstances/blob/bc935ac944470efc074cae922ce6cdaf6709ae20/src/Data/NumInstances/Tuple.hs#L18-L30.